eth/downloader/downloader_test.go

```diff
@@ -620,15 +499,13 @@ func testThrottling(t *testing.T, protocol uint, mode SyncMode) {
 	if err := <-errc; err != nil {
 		t.Fatalf("block synchronization failed: %v", err)
 	}
-	tester.terminate()
-
 }
 
 // Tests that simple synchronization against a forked chain works correctly. In
 // this test common ancestor lookup should *not* be short circuited, and a full
 // binary search should be executed.
 func TestForkedSync66Full(t *testing.T)  { testForkedSync(t, eth.ETH66, FullSync) }
-func TestForkedSync66Fast(t *testing.T)  { testForkedSync(t, eth.ETH66, FastSync) }
+func TestForkedSync66Snap(t *testing.T)  { testForkedSync(t, eth.ETH66, SnapSync) }
```

Comment:
Yes, this is unfortunately caused by all the tests running in parallel and too many generating chains. The downloader has built in timeouts that dynamically adjust to peers. If the downloader starts at one speed, and then another test spins up and hammers the CPU, the first one will be starved (compared to previously). Starvation will lead to the serving peer timing out and failing the test due to "stalling".

I can try to add a pre-generation step similar to what Felix did instead of the current lazy one. That would hopefully solve it, it's just wonky.

Reply:
I made some changes to pregenerate the datasets and run the tests sequentially. Also reduced the dataset sizes a bit. Hopefully will stop dropping dead.